yarpl/include/yarpl/flowable/Subscriber.h

```diff
@@ -15,40 +15,119 @@ namespace flowable {
 template <typename T>
 class Subscriber : public virtual Refcounted, public yarpl::enable_get_ref {
  public:
+  virtual void onSubscribe(Reference<Subscription>) = 0;
+  virtual void onComplete() = 0;
+  virtual void onError(folly::exception_wrapper) = 0;
+  virtual void onNext(T) = 0;
+};
+
+// codemod all things that inherit from Subscriber to inherit from
+// InternalSubscriber
+template <typename T>
+class InternalSubscriber : public Subscriber<T> {
```

Comment:
Do we ever need this class?
What it does it only keeping a `subscription_` and just calls `reset` on this object. It is one line of code that the class which extends this one can do easily. 
Do we see that we are going to have any instance of this class at all?
So just comments in the interface should suffice, so we would not force the implementers of the interface to have `subscription_` object be in the parent class.

Reply:
`InternalSubscriber` is verbatim what `Subscriber` used to be

Reply:
Can we get rid of the InternalSubscriber and use the old Subscriber instead? In the next diff we can remove the subscription pointer from the Subscriber and fix the places which were using it.